File: 0268-missing-number/0268-missing-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function (nums) {
    let n = nums.length;
    let totalSum = (n * (n + 1)) / 2;

    let currSum = 0;
    for (let i = 0; i < nums.length; i++) {
        currSum += nums[i];
    }

    return totalSum - currSum;
};

File: 0268-missing-number/0268-missing-number.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)
        totalSum = (n * (n + 1)) // 2
        currSum = 0

        for i in range(n):
            currSum += nums[i]

        return totalSum - currSum
